start-tests.sh

```diff
@@ -7,106 +7,267 @@
-  export CYPRESS_OPTIONS_MANAGED_CLUSTER_NAME=`yq e '.options.clusters[0].name' $USER_OPTIONS_FILE`
   export OPTIONS_HUB_BASEDOMAIN=`yq e '.options.hub.baseDomain' $USER_OPTIONS_FILE`
   export OPTIONS_HUB_USER=`yq e '.options.hub.user' $USER_OPTIONS_FILE`
   export OPTIONS_HUB_PASSWORD=`yq e '.options.hub.password' $USER_OPTIONS_FILE`
+  export OPTIONS_HUB_KUBECONTEXT=`yq e '.options.hub.kubecontext' $USER_OPTIONS_FILE`
+  export OPTIONS_MANAGED_BASEDOMAIN=`yq e '.options.clusters[0].basedomain' $USER_OPTIONS_FILE`
   export OPTIONS_MANAGED_CLUSTER_NAME=`yq e '.options.clusters[0].name' $USER_OPTIONS_FILE`
+  export OPTIONS_MANAGED_KUBECONFIG=`yq e '.options.clusters[0].kubeconfig' $USER_OPTIONS_FILE`
+  export OPTIONS_HUB_OC_IDP=`yq e '.options.identityProvider' $USER_OPTIONS_FILE`
 else
-  log_color "yellow" "Options file does not exist, using test config from environment variables.\n"
+  log_color "yellow" "Options file does not exist, checking to see if the test can be configured with environment variables."
 fi
 
-export CYPRESS_BASE_URL=https://multicloud-console.apps.$CYPRESS_OPTIONS_HUB_BASEDOMAIN
+# Check to see if OPTIONS_HUB_OC_IDP is unset or null.
+if [[ -z $OPTIONS_HUB_OC_IDP || "$OPTIONS_HUB_OC_IDP" == "null" ]]; then
+  log_color "purple" "OPTIONS_HUB_OC_IDP" "is (null or not set); setting to 'kube:admin'\n"
+  export OPTIONS_HUB_OC_IDP=kube:admin
+else
+  log_color "purple" "OPTIONS_HUB_OC_IDP" "detected, using $OPTIONS_HUB_OC_IDP for test.\n"
+fi
 
-# Check to see if CYPRESS_OC_IDP is null.
-if [[ -z $CYPRESS_OC_IDP || "$CYPRESS_OC_IDP" == "null" ]]; then
-  log_color "purple" "CYPRESS_OC_IDP" "is (null or not set); setting to 'kube:admin'\n"
-  export CYPRESS_OC_IDP=kube:admin
+# Create dir for kube configs.
+mkdir -p ./kube/config
+
+# Check to see if OPTIONS_HUB_BASEDOMAIN, OPTIONS_HUB_USER, or OPTIONS_HUB_PASSWORD are missing. We need these to run the UI test.
+if [[ -z $OPTIONS_HUB_BASEDOMAIN || -z $OPTIONS_HUB_USER || -z $OPTIONS_HUB_PASSWORD ]]; then
+  log_color "red" "One or more exported variables are undefined for hub cluster." "(set ${PURPLE}OPTIONS_HUB_BASEDOMAIN, OPTIONS_HUB_BASEDOMAIN, and OPTIONS_HUB_BASEDOMAIN${NC} to execute the test with environment variables)\n"
+
+  # Check to see if the kubeconfig for the hub cluster is available.
+  if [[ ! -f $OPTIONS_HUB_KUBECONFIG ]]; then
+    log_color "red" "The kubeconfig file for the hub cluster was not located." "(set ${PURPLE}KUBECONFIG${NC} to ${YELLOW}$OPTIONS_HUB_KUBECONFIG${NC} and oc login to create kubeconfig file)"
+    exit 1
+  else
+    # To run the E2E test, we need the password to log into the UI. If it's not exported or available from the options.yaml file, skip the UI test.
+    if [[ -z $OPTIONS_HUB_PASSWORD || "$OPTIONS_HUB_PASSWORD" == "null" ]]; then
+      log_color "purple" "OPTIONS_HUB_PASSWORD" "is required to login into the ACM console. ${YELLOW}Skipping UI test.${NC}\n"
+      export SKIP_UI_TEST=true
```

Comment:
We need the `OPTIONS_HUB_PASSWORD` to login into the UI. If we do not have that variable exported, skip the `ui` test and only run the API test. We can run the API test if the `kubeconfig` context is available for us.